force-app/main/default/classes/PMT_RefundController.cls

```diff
@@ -57,8 +57,12 @@ public with sharing class PMT_RefundController {
         refundService.withOriginalPaymentIds(new Set<Id>{paymentId});
         PMT_RefundService.RefundInfo refundInfo = refundService.getRefundInfoFor(paymentId);
 
-        refundView.paymentDate = refundInfo.originalPayment.npe01__Payment_Date__c;
-        refundView.remainingBalance = refundInfo.remainingBalance;
+        if (isElevatePayment(refundInfo.originalPayment.Elevate_Payment_ID__c)) {
```

Comment:
`refundService.getRefundInfoFor(paymentId);` did the query to retrieve the data. It will always query the record and assign to refundInfo. However, it's still good to have null check in place just in case. The safe navigation operator is added.